gsw_xarray/_core.py

```diff
@@ -36,55 +38,93 @@ def quantify(rv, attrs, unit_registry=None):
     return rv
 
 
-def pint_compat(args, kwargs):
+def dequantify_reg(kw, arg):
```

Comment:
This almost feels like a good candidate for functools.singledispatch, though that uses the type of the first argument

Reply:
I may be possible to use singledispatch but I am not sure that we would gain a lot here.

I had another idea (but not very precise) that I wanted to (try to) implement after this PR: refactor the pint_compat, add_attr, quantify, etc into decorators, so the code would look like:
```python
@pint_compat
@dequantify
def funcwrap(func, args, kwargs):
    # call func(args, kwargs)
    pass
```
I guess we have to split out functions into 2 groups: functions working before calling gsw.function - dequantify, pint_compat - and fucntions working after - quantify, add_attrs -

Reply:
Sorry I think that I missed your point... I thought that you were proposing to use functools.singledispatch for the wrapper.
It seems that it is well appropriated for `dequantify_reg`

Reply:
done